src/components/mypage/ReviewModal/index.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from 'react'
+import { Styled } from './styled'
+import type {
+  WriteReviewModalProps,
+  ReadReviewModalProps,
+  ScoreState,
+  SCORE,
+  ReviewState,
+  CommonReviewModalProps
+} from './types'
+
+const MOCK_SCORE: SCORE = [
+  {
+    state: 'smile',
+    text: '좋아요'
+  },
+  {
+    state: 'meh',
+    text: '보통이에요'
+  },
+  {
+    state: 'sad',
+    text: '별로에요'
+  }
+]
+
+const CommonTitleContainer = ({
+  nickname,
+  productName
+}: CommonReviewModalProps): ReactElement => (
+  <Styled.TitleContainer>
+    <Styled.FirstSection>
+      <Styled.NickName>{nickname}</Styled.NickName>
+      <Styled.NormalText>님과</Styled.NormalText>
+    </Styled.FirstSection>
+    <Styled.NormalText>거래는 어땠나요?</Styled.NormalText>
+    <Styled.ProductText>{productName}</Styled.ProductText>
+  </Styled.TitleContainer>
+)
+
+const WriteReviewModal = ({
+  isOpen = true,
+  onClose,
+  onConfirm,
+  nickname = '닉네임',
+  productName = '상품이름'
+}: WriteReviewModalProps): ReactElement => {
+  const [isClickReviewIcon, setIsClickReViewIcon] = useState<boolean>(false)
+
+  const [reviewState, setReviewState] = useState<ReviewState>({
+    reviewScore: null,
+    reviewText: ''
+  })
+
+  const handleClickReviewIcon = (reviewScore: ScoreState): void => {
+    setIsClickReViewIcon(true)
+    setReviewState({ ...reviewState, reviewScore })
+  }
+
+  const handleInput: ChangeEventHandler<HTMLTextAreaElement> = e => {
+    setReviewState({
+      ...reviewState,
+      reviewText: e.target.value
+    })
+  }
+
+  return (
+    <Styled.ReviewModal isOpen={isOpen} onClose={onClose}>
+      <CommonTitleContainer
+        nickname={nickname}
+        productName={productName}></CommonTitleContainer>
+      <Styled.ReviewIconContainer>
+        {MOCK_SCORE.map(scoreItem => {
+          return (
+            <Styled.ReviewState
+              key={scoreItem.state}
+              onClick={(): void => handleClickReviewIcon(scoreItem.state)}>
+              <Styled.ReviewIcon
+                isFill={reviewState.reviewScore === scoreItem.state}
+                isGood={scoreItem.state === 'smile'}
+                type={
+                  reviewState.reviewScore === scoreItem.state
+                    ? `${scoreItem.state}Fill`
+                    : scoreItem.state
+                }></Styled.ReviewIcon>
+              <Styled.ReviewText
+                isFill={reviewState.reviewScore === scoreItem.state}>
+                {scoreItem.text}
+              </Styled.ReviewText>
+            </Styled.ReviewState>
+          )
+        })}
+      </Styled.ReviewIconContainer>
+
+      <Styled.ReviewTextArea
+        guideMessage={`${reviewState.reviewText?.length}/100`}
+        maxLength={100}
+        onInput={handleInput}></Styled.ReviewTextArea>
+
+      <Styled.ReviewSendButton
+        disabled={!isClickReviewIcon}
+        styleType={isClickReviewIcon ? 'solidPrimary' : 'solidDisabled'}
+        onClick={(): void => onConfirm(reviewState)}>
```

Comment:
```suggestion
const handleConfirm = () => {
  onConfirm(reviewState)
}

        onClick={handleConfirm}>
```
효정님 코멘트에도 남겨두었는데 함수 전,후 처리 없는 경우에는 인라인으로 적고 지금처럼 reviewState를 넣어서 호출해야하는 전 처리가 있는 경우 핸들러 함수로 분리하는 것이 추후 유지보수할 때 용이할 것 같은데 어떻게 생각하시나요??

Reply:
넵 너무 동의합니다 
효정님 코멘트도 확인하겠습니다!

Reply:
완롼~
[링크](https://github.com/price-offer/offer-fe/pull/108/commits/fee8074b8f9f054f36734eeda9c8b6bd81a57cc9#diff-efe33139ad392b5168600bf34f17b3e692070a8d4f1bb8f20bc10c2f0ba7ab16)